lib/executor.js

```diff
@@ -305,12 +306,18 @@ function runScripts(app, list, callback) {
 
   async.eachSeries(functions, function(f, done) {
     debug('Running script %s', f.path);
-    if (f.func.length >= 2) {
+    var isAsync = utils.isAsync(f.func);
+    if (f.func.length >= 2 || isAsync) {
       debug('Starting async function %s', f.path);
-      f.func(app, function(err) {
+      var cb = function(err) {
         debug('Async function finished %s', f.path);
         done(err);
-      });
+      };
+      if (isAsync) {
+        f.func(app).then(cb.bind(null, null)).catch(cb);
```

Comment:
If `cb(null)` throws an error, then `.catch()` will invoke `cb` again with this error - that's not correct IMO. You should pass the error-handling-callback as the second argument of `.then()`:

```js
f.func(app).then(cb.bind(null, null), cb);
```